core/src/main/java/gyro/core/diff/Diff.java

```diff
@@ -279,6 +285,21 @@ private Change newDelete(Diffable diffable) {
         return delete;
     }
 
+    private void setCredentials(Diffable diffable) {
```

Comment:
This name should be more descriptive, `setCredentials` implies a normal setter but this appears to be searching around in state for the appropriate credentials to use to delete.